src/components/ImageView/Colorbar/ColorbarComponent.tsx

```diff
@@ -99,28 +141,57 @@ export class ColorbarComponent extends React.Component<ColorbarComponentProps> {
                 if (colorbarSettings.tickVisible) {
                     // to avoid blurry ticks when width <= 1px, offset to .5 px position 
                     const position = positions[i] - ((colorbarSettings.tickWidth <= 1) && (positions[i] - Math.floor(positions[i]) - 0.5 / devicePixelRatio));
+                    let tickPoints = [colorbarSettings.rightBorderPos - colorbarSettings.tickLen, position, colorbarSettings.rightBorderPos, position];
+                    if (colorbarSettings.position !== "right") { // rotate to horizontal by swapping
+                        tickPoints = [tickPoints[1], tickPoints[0], tickPoints[3], tickPoints[2]];
+                        if (colorbarSettings.position === "top") { // reflect over x-axis
+                            tickPoints[1] = colorbarSettings.rightBorderPos + colorbarSettings.tickLen;
+                        }
+                    }
                     ticks.push(
                         <Line
-                            points={[colorbarSettings.rightBorderPos - colorbarSettings.tickLen, position, colorbarSettings.rightBorderPos, position]}
+                            points={tickPoints}
                             stroke={getColor(colorbarSettings.tickCustomColor, colorbarSettings.tickColor)}
                             strokeWidth={colorbarSettings.tickWidth / devicePixelRatio}
                             key={i.toString()}
                         />
                     );
                 }
                 if (colorbarSettings.numberVisible) {
+                    let numberXPos = colorbarSettings.rightBorderPos + colorbarSettings.textGap;
+                    let numberYPos = positions[i] - colorbarSettings.height / 2;
+                    if (colorbarSettings.position !== "right") { // rotate to horizontal by swapping
+                        [numberXPos, numberYPos] = [numberYPos, numberXPos];
+                        if (colorbarSettings.position === "top") {
+                            numberYPos = colorbarSettings.rightBorderPos - colorbarSettings.textGap - colorbarSettings.numberFontSize;
+                        }
+                    } else { // adjust for rotation
+                        switch(colorbarSettings.numberRotation) {
```

Comment:
need to follow coding style:
1. blank after "switch"
2. no "()" in case statement

Reply:
Changed the style.